### PR TITLE
implemented issues 16 and 19

### LINKED ISSUE #16: OOP: Implement methods for class in_out
Implement all methods for this class and document parameters

### LINKED ISSUE #19: OOP: Implement method "calc eucl dist" of class "analysis"
Implement method

### PR DESCRIPTION
**Pull request by vivienzahs**
- Implemented issues 16 and 19.
- Added suggestion for how to handle user input (currently commented out)

Closed issues:

- Closes #16 
- Closes #19

### CODE REVIEW

File: src/oop_analysis.py [hunk 1]
```diff
@@ -37,31 +72,96 @@ def main(argv):
...
+        self.file_out = df.to_csv(file_out, header=header)
 
 
 class analysis:
-    def __init__(self, df: str):
+    def __init__(self, df: str, table_name: str):
+        self.table = None
         self.df = df
         self.df_filtered = None
+        self.table_name = table_name
```
Comment (han16nah) [inline src/oop_analysis.py:136]: not sure if this one is needed

**Review by han16nah — CHANGES_REQUESTED** [review]
I added some comments to sections that may be changed.

Let's decide tomorrow whether to use ReStructuredText (like you did here) or Google format (like I did in #21) for the docstrings

Add some point, we should add type hints (in a new issue/PR, once we implemented rsmaining methods), e.g. 
```python
def read_csv_as_df(self, file_in: str, col_separator: str = " ", skip_initial_space: bool = True) -> None:
```

File: src/oop_analysis.py [hunk 2]
```diff
@@ -13,20 +17,51 @@
...
+            print('col_sep is "', col_sep)
+        elif opt in ("-skip_init_space", "--skip_init_space"):
+            skip_init_space = arg
+            print('skip_init_space "', skip_init_space)
+        elif opt in ("-file_out", "--file_out"):
+            file_out = arg
+            print('file_out is"', file_out)
+        elif opt in ("-header_outfile", "--header_outfile"):
+            header_outfile = arg
+            print('header_outfile is"', header_outfile)
```
[the remark this answers is not in the document]
Comment (vivienzahs) [reply src/oop_analysis.py:62]: Yeah, I think its is only two different modules doing similar things. Found this quite nice, but is still commented out and open for discussion. I am fine with either modules.

**Comment by vivienzahs** [comment]
I have included all the changes and have solved the conflicts. I think this branch can now be merged with the main branch.

**Pull request merged**